src/typings.ts

```diff
@@ -14,14 +14,20 @@ interface BaseViewModel {
 
 // View-model interface with reactive Props from React component
 export type ViewModelProps = {};
+export type ViewModelCtx = Array<unknown>
 export interface ViewModelWithProps<P extends ViewModelProps>
   extends BaseViewModel {
   // You should use @observable.ref for props in your view-model classes
   props: P;
 }
 
-// Класс-конструктор для view-моделей
+// Constructor class for view models
 export type IViewModelClass<
   T extends IViewModel<P>,
-  P extends ViewModelProps | undefined = undefined
-> = P extends {} ? { new (props: P): T } : { new (): T };
+  P extends ViewModelProps | undefined = undefined,
+  CTX extends ViewModelCtx = ViewModelCtx
```

Comment:
Are you sure that `Ctx` is a good naming for constructor arguments? Maybe we name it `Args` after all?